### PR TITLE
Port EE fix to disposal chutes

### PR DESCRIPTION
**Pull request by Lyndomen**
## About the PR
Disposal chutes shoot at orientation 0 no matter where they are pointing at now on the grid as i detailed in https://github.com/space-wizards/space-station-14/issues/31459 

I would love to PR this upstream but AGPL :trollface: 

ported from https://github.com/Fansana/floofstation1/pull/220
> This PR fixes an issue with disposal pipes that lead into space. The ejection would use the transform of the map instead of the one from the grid and throw into the wrong direction.
> Another issue this fixes is with disposal bends which would take direction from a previous pipe instead of the current one.

## Why / Balance
Important for https://github.com/DeltaV-Station/Delta-v/pull/1582 and any ship with a disposal cannon 

## Technical details
tiny 2 line change C#

## Media
image

## Requirements
- [X] I have tested all added content and changes.
- [X] I have added media to this PR or it does not require an ingame showcase.

## Breaking changes
no
**Changelog**

:cl:
- fix: Disposal chutes on a shuttle will now shoot out in the appropriate direction

### CODE REVIEW

File: Content.Server/Disposal/Unit/EntitySystems/DisposableSystem.cs [hunk 1]
```diff
@@ -136,9 +136,9 @@ public void ExitDisposals(EntityUid uid, DisposalHolderComponent? holder = null,
                 {
                     _xformSystem.AttachToGridOrMap(entity, xform);
 
-                    if (holder.PreviousDirection != Direction.Invalid && _xformQuery.TryGetComponent(xform.ParentUid, out var parentXform))
+                    if (holder.PreviousDirection != Direction.Invalid && gridUid != null && _xformQuery.TryGetComponent(gridUid, out var parentXform))  // DeltaV begin EE Disposal fix
                     {
-                        var direction = holder.PreviousDirection.ToAngle();
+                        var direction = holder.CurrentDirection.ToAngle(); // DeltaV end EE Disposal fix
```
Comment (deltanedas) [inline Content.Server/Disposal/Unit/EntitySystems/DisposableSystem.cs]: end

File: Content.Server/Disposal/Unit/EntitySystems/DisposableSystem.cs [hunk 2]
```diff
@@ -136,9 +136,9 @@ public void ExitDisposals(EntityUid uid, DisposalHolderComponent? holder = null,
                 {
                     _xformSystem.AttachToGridOrMap(entity, xform);
 
-                    if (holder.PreviousDirection != Direction.Invalid && _xformQuery.TryGetComponent(xform.ParentUid, out var parentXform))
+                    if (holder.PreviousDirection != Direction.Invalid && gridUid != null && _xformQuery.TryGetComponent(gridUid, out var parentXform))  // DeltaV begin EE Disposal fix
```
Comment (deltanedas) [inline Content.Server/Disposal/Unit/EntitySystems/DisposableSystem.cs]: why does this need to use the grids world rotation, the disposal unit should be the same as the grid

**Review by deltanedas — CHANGES_REQUESTED** [review]

File: Content.Server/Disposal/Unit/EntitySystems/DisposableSystem.cs [hunk 3]
```diff
@@ -136,9 +136,9 @@ public void ExitDisposals(EntityUid uid, DisposalHolderComponent? holder = null,
                 {
                     _xformSystem.AttachToGridOrMap(entity, xform);
 
-                    if (holder.PreviousDirection != Direction.Invalid && _xformQuery.TryGetComponent(xform.ParentUid, out var parentXform))
+                    if (holder.PreviousDirection != Direction.Invalid && _xformQuery.TryGetComponent(uid, out var parentXform))  // start DeltaV EE Disposal fix
```
Comment (deltanedas) [inline Content.Server/Disposal/Unit/EntitySystems/DisposableSystem.cs:139]: just leave it how it was

Comment (Lyndomen) [reply Content.Server/Disposal/Unit/EntitySystems/DisposableSystem.cs:139]: the item shot just dissapears if i dont do this change or the one prev committed

Comment (deltanedas) [reply Content.Server/Disposal/Unit/EntitySystems/DisposableSystem.cs:139]: the pipe's WorldRotation and the parent (grid) Should be the exact same, are you sure they are different

Comment (Lyndomen) [reply Content.Server/Disposal/Unit/EntitySystems/DisposableSystem.cs:139]: I'm not sure of anything but when I changed it to only edit on line  141, no edit on 139, the objects sit in one place then eventually disappear

**Comment by Lyndomen** [comment]
Closed, being resolved upstream

**Pull request closed without merging**